src/sources/aws_ecs_metrics/parser.rs

```diff
@@ -168,86 +168,102 @@ fn blkio_metrics(
-            "recursive_io_service_bytes_total",
-            namespace.clone(),
-            timestamp,
-            s.value,
-            blkio_tags(s, tags),
-        )
-    }));
-    metrics.extend(blkio.io_service_time_recursive.iter().map(|s| {
-        counter(
-            "blkio",
-            "recursive_io_service_time_seconds_total",
-            namespace.clone(),
-            timestamp,
-            s.value / 1_000_000_000.0,
-            blkio_tags(s, tags),
-        )
-    }));
-    metrics.extend(blkio.io_serviced_recursive.iter().map(|s| {
-        counter(
-            "blkio",
-            "recursive_io_serviced_total",
-            namespace.clone(),
-            timestamp,
-            s.value,
-            blkio_tags(s, tags),
-        )
-    }));
-    metrics.extend(blkio.io_time_recursive.iter().map(|s| {
-        counter(
-            "blkio",
-            "recursive_io_time_seconds_total",
-            namespace.clone(),
-            timestamp,
-            s.value / 1_000.0,
-            blkio_tags(s, tags),
-        )
-    }));
-    metrics.extend(blkio.io_wait_time_recursive.iter().map(|s| {
-        counter(
-            "blkio",
-            "recursive_io_wait_time_seconds_total",
-            namespace.clone(),
-            timestamp,
-            s.value / 1_000_000_000.0,
-            blkio_tags(s, tags),
-        )
-    }));
-    metrics.extend(blkio.sectors_recursive.iter().map(|s| {
-        counter(
-            "blkio",
-            "recursive_sectors_total",
-            namespace.clone(),
-            timestamp,
-            s.value,
-            blkio_tags(s, tags),
-        )
-    }));
+    if let Some(io_merged_recursive) = &blkio.io_merged_recursive {
+        metrics.extend(io_merged_recursive.iter().map(|s| {
+            counter(
+                "blkio",
+                "recursive_io_merged_total",
+                namespace.clone(),
+                timestamp,
+                s.value,
+                blkio_tags(s, tags),
+            )
+        }));
+    }
+    if let Some(io_queue_recursive) = &blkio.io_queue_recursive {
+        metrics.extend(io_queue_recursive.iter().map(|s| {
+            counter(
+                "blkio",
+                "recursive_io_queued_total",
+                namespace.clone(),
+                timestamp,
+                s.value,
+                blkio_tags(s, tags),
+            )
+        }));
+    }
+    if let Some(io_service_bytes_recursive) = &blkio.io_service_bytes_recursive {
+        metrics.extend(io_service_bytes_recursive.iter().map(|s| {
+            counter(
+                "blkio",
+                "recursive_io_service_bytes_total",
+                namespace.clone(),
+                timestamp,
+                s.value,
+                blkio_tags(s, tags),
+            )
+        }));
+    }
+    if let Some(io_service_time_recursive) = &blkio.io_service_time_recursive {
+        metrics.extend(io_service_time_recursive.iter().map(|s| {
```

Comment:
Nit: I think what you have here is fine, but given `Option` is an iterator too I think you can write this as:

```
metrics.extend(blkio.io_service_time_recursive.flatten().iter().map(|s| {
```

Reply:
Ah! I was looking for something that would be simpler than introducing an `if let` , thanks for the tip!